src/usermeta.ts

```diff
@@ -587,6 +690,112 @@ class UsermetaRouter {
+      const clientType = client.type;
+      if (!clients[clientType]) {
+        clients[clientType] = {};
+        clients[clientType].keys = [];
+        clients[clientType].client = client;
+      }
+      clients[clientType].keys.push(key);
+    });
+
+    let tasks: any[] = [];
+    //generate the tasks for parallel
+    Object.keys(clients).forEach((clientType: string) => {
+      const clientObj = clients[clientType];
+      const _keys: string[] = clientObj.keys;
+      const _client = clientObj.client;
+      tasks = tasks.concat(buildTask(_client, _keys));
+    });
+    return tasks;
+  }
+
+  getBulk(params: UsermetaClientBulkOptions | string, keys: string[], cb: UsermetaClientBulkCallback) {
+    //generate tasks for get.
+    if (keys.length === 0)
+      return cb(null, []);
+    const tasks: any[] = this.getTasksForBulk(params, keys, (_client, _keys) => {
+      let _tasks: any[] = [];
+      //if getBulk method exists on this client ..  then we call getbulk method => and passing the keys
+      //related.
+      if (_client.getBulk) {
+        _tasks.push(cb1 => _client.getBulk(params, _keys, cb1));
+      }
+      else {
+        //else we loop over the keys => and we will create tasks based for each key => client =>call for key.
+        _keys.forEach((key: string) => {
+          _tasks.push(cb1 => _client.get(params, key, (err: Error | null, reply?: string | null) => {
+            if (err) {
+              return cb1(err, reply);
+            }
+            cb1(err, { key, value: reply });
+          }));
+        });
+      }
+      return _tasks;
+    });
+
+    async.parallel(tasks, (err, data) => {
+      if (err) {
+        cb(err, null);
+      }
+      else {
+        //make the 2 levels array to 1 level => [[], [], []] => []
+        let oneLevelArray = (data as any[][])?.flat();
+        cb(null, oneLevelArray);
+      }
+    });
+  }
+  setBulk(pparams: UsermetaClientBulkOptions | string, keyValues: KeyValue[], cb: UsermetaClientBulkCallback) {
+    if (keyValues.length === 0)
+      return cb(null, 'OK');
+    //get list of keys from keyvalue pair.
+    const keys = keyValues.map((i) => i.key);
+    //generate tasks for set.
+    const tasks: any[] = this.getTasksForBulk(pparams, keys, (_client, _keys) => {
+      let _tasks: any[] = [];
+      // if setBulk method exists on this client ..  then we call setbulk method => and passing the keys
+      // and valuesrelated.
+      if (_client.setBulk) {
+        const values = keyValues.filter((x) => _keys.includes(x.key));
+        _tasks.push(cb1 => _client.setBulk(pparams, values, cb1));
+      }
+      else {
+        //else we loop over the keys => and we will create tasks based for each key => client =>call for key/value.
+        _keys.forEach((key: string) => {
+          const val = keyValues.filter((x) => x.key == key)[0].value;
+          _tasks.push(cb1 => _client.set(pparams, key, val, cb1));
+        });
+      }
+      return _tasks;
```

Comment:
same remark, no need to create a variable called `_tasks` (see similar code above)